bootloader/src/pyi_archive.c

```diff
@@ -377,6 +378,145 @@ findDigitalSignature(ARCHIVE_STATUS * const status)
 #endif /* ifdef _WIN32 */
 }
 
+/*
+ * Try to find the pydata section in an ELF executable.
+ * Returns the end of the section if found.
+ */
+static int
+findElfSection(ARCHIVE_STATUS * const status)
+{
+#if !defined(WIN32) && !defined(__APPLE__)
+    char head[64];
+    uint16_t endianess_bytes = 0x0201;
```

Comment:
Isn't endianness a single byte?

Reply:
@lkollar @htgoebel Reading one byte of a multibyte int is a common runtime check for endianess.

The EI_DATA field in the ELF header is a byte that indicates little (value=1) or big (value=2) endian. Putting `endianess_bytes` with the 16bit value 0x0201 on the stack and reading its first byte yields 0x01 on little and 0x02 on big endian systems.

We can #define the value elsewhere but we have to put it into an addressable memory location here to access a single byte.

Reply:
@durko sorry, totally missed this one. Feel free to dismiss my comments around the endianness check.

Reply:
Regarding the whole endiness checks: I'm in favor of less "magic", thus someone less skill in C (like me ;-) is able to understand the code easily.

I suggest using more verbose code, something along these lines:
```
// these values need to match ELF header flags
#define BIG_ENDIAN 2    # or ENDIAN_BIG, I don't care
#define LITTLE_ENDIAN 1
// commonly used way to determine CPU endianess: Put uint16 onto the
// stack and read the first byte to learn whether it is the upper or lower byte.
uint16_t endianess_bytes = BIG_ENDIAN <<8 + LITTLE_ENDIAN;
uint8_t endianess = *(uint8_t*)&endianess_bytes;
…
if (head[0x05] != endianess {
…
```